packages/cached/test/cached-fs.spec.ts

```diff
@@ -11,279 +11,419 @@ describe('createCachedFs', () => {
+            const filePath = '/file';
+            const memFs = createMemoryFs({ [filePath]: SAMPLE_CONTENT });
+      
+            const readFileSyncSpy = sinon.spy(memFs, 'readFileSync');
+      
+            const fs = createCachedFs(memFs);
+      
+            fs.readFileSync(filePath);
+            fs.invalidate(filePath);
+            fs.readFileSync(filePath);
+      
+            expect(readFileSyncSpy.callCount).to.equal(2);
+        });
+
+        it('allows invalidating cache of non existing file path', async () => {
+            const filePath = '/file';
+            const memFs = createMemoryFs({ [filePath]: SAMPLE_CONTENT });
+      
+            const readFileSyncSpy = sinon.spy(memFs, 'readFileSync');
+      
+            const fs = createCachedFs(memFs);
+      
+            try {
+              fs.readFileSync(filePath);
+            } catch (ex) {
+              // NO-OP
+            }
+            fs.invalidate(filePath);
+            try {
+              fs.readFileSync(filePath);
+            } catch (ex) {
+              // NO-OP
+            }
+      
+            expect(readFileSyncSpy.callCount).to.equal(2);
+        });
+      
+        it('allows invalidating cache of all file paths', async () => {
+            const filePath = '/file';
+            const memFs = createMemoryFs({ [filePath]: SAMPLE_CONTENT });
+      
+            const readFileSyncSpy = sinon.spy(memFs, 'readFileSync');
+      
+            const fs = createCachedFs(memFs);
+      
+            fs.readFileSync(filePath);
+            fs.invalidateAll();
+            fs.readFileSync(filePath);
+      
+            expect(readFileSyncSpy.callCount).to.equal(2);
+        });
+
+        it('caches readfile (callback-style) calls - file exists', async () => {
+            const filePath = '/file';
+            const memFs = createMemoryFs({ [filePath]: SAMPLE_CONTENT });
+      
+            const readFileSpy = sinon.spy(memFs, 'readFile');
+      
+            const fs = createCachedFs(memFs);
+      
+            const content = await new Promise((res, rej) =>
+              fs.readFile(filePath, (error, value) => (error ? rej(error) : res(value)))
+            );
+      
+            const content2 = await new Promise((res, rej) =>
+              fs.readFile(filePath, (error, value) => (error ? rej(error) : res(value)))
+            );
+      
+            expect(content).to.equal(content2);
+            expect(readFileSpy.callCount).to.equal(1);
+        });
+      
+        it('caches readfile (callback-style) calls - file does not exist', async () => {
+            const filePath = '/file';
+            const memFs = createMemoryFs({ [filePath]: SAMPLE_CONTENT });
+      
+            const readFileSpy = sinon.spy(memFs, 'readFile');
+      
+            const fs = createCachedFs(memFs);
+      
+            try {
+              await new Promise((res, rej) => fs.readFile('/no-file', (error, value) => (error ? rej(error) : res(value))));
```

Comment:
since the rejections here are no-op, could just `await new Promise((res) => fs.readFile('/no-file', res))` and remove the try/catch

Reply:
fixed